packages/web-scripts/src/allSettled.ts

```diff
@@ -0,0 +1,36 @@
+interface PromiseResult {
+  status: string;
+  reason?: Error;
+  value?: any;
+}
+interface PromiseRejection extends PromiseResult {
+  status: 'rejected';
+  reason: Error;
+}
+interface PromiseFulfillment<T> extends PromiseResult {
+  status: 'fulfilled';
+  value: T;
+}
+
+export function allSettled<T>(arr: Promise<T>[]): Promise<PromiseResult[]> {
+  return Promise.all(
+    arr.map(
+      promise =>
+        new Promise(resolve => {
+          promise.then(
+            (value: T) => {
+              const result: PromiseFulfillment<T> = {
+                status: 'fulfilled',
+                value,
+              };
+              resolve(result);
+            },
+            (reason: Error) => {
+              const result: PromiseRejection = { status: 'rejected', reason };
+              resolve(result);
+            },
+          );
+        }),
+    ),
+  );
+}
```

Comment:
I tried to use https://www.npmjs.com/package/promise.allsettled but it was not working very well for me. this is a simple implementation of https://github.com/tc39/proposal-promise-allSettled, which is in stage 4 but doesn't have wide adoption yet it seems.